test/tests/functional/pbs_acl_host_moms.py

```diff
@@ -94,13 +97,68 @@ def test_acl_host_moms_enable(self):
         ret = self.du.run_cmd(self.remote_host, cmd=self.pbsnodes_cmd)
         self.assertEqual(ret['rc'], 0)
 
+        ret = self.du.run_cmd(self.remote_host, cmd=self.qstat_cmd)
+        self.assertEqual(ret['rc'], 0)
+
     def test_acl_host_moms_disable(self):
         """
         Set acl_host_moms_enable = False and check whether or not the remote
-        host is forbidden to run pbsnodes.
+        host is forbidden to run pbsnodes and qstat.
         """
         self.server.manager(MGR_CMD_SET, SERVER, {
                             'acl_host_moms_enable': False}, expect=True)
 
         ret = self.du.run_cmd(self.remote_host, cmd=self.pbsnodes_cmd)
-        self.assertEqual(ret['rc'], 1)
+        self.assertNotEqual(ret['rc'], 0)
+
+        ret = self.du.run_cmd(self.remote_host, cmd=self.qstat_cmd)
+        self.assertNotEqual(ret['rc'], 0)
+
+    def test_acl_host_moms_hooks_and_jobs(self):
+        """
+        Use hooks to test whether remote host is able to run pbs.server()
+        and check whether the job that is submitted goes to the 'R' state.
+        """
+        hook_name = "hook_acl_host_moms"
+        hook_body = """
+import pbs
+e = pbs.event()
+svr = pbs.server()
+e.accept()
+"""
+        a = {'event': 'queuejob', 'enabled': 'True'}
+        rv = self.server.create_import_hook(
+            hook_name, a, hook_body, overwrite=True)
+        self.assertTrue(rv)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {
+                            'acl_host_moms_enable': True}, expect=True)
+
+        qsub_cmd_sleep = os.path.join(self.server.pbs_conf[
+            'PBS_EXEC'], 'bin', 'qsub') + ' -- /bin/sleep 10'
+        ret = self.du.run_cmd(self.remote_host, cmd=qsub_cmd_sleep)
+        self.assertEqual(ret['rc'], 0)
+        jid = ret['out'][0]
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+
+    def test_acl_host_mom_queue_access(self):
+        """
+        Check whether remote host is able to access queues where
+        acl_host_enable is True.
```

Comment:
Test looks like testing that remote host is not able to access queue when acl_host_moms_enable is true and acl_hosts not set on queue . Please update test description in comments